homeassistant/components/opower/coordinator.py

```diff
@@ -85,20 +85,50 @@ async def _async_update_data(
         # Because Opower provides historical usage/cost with a delay of a couple of days
         # we need to insert data into statistics.
         await self._insert_statistics()
-        return {forecast.account.utility_account_id: forecast for forecast in forecasts}
+        return {forecast.account.uuid: forecast for forecast in forecasts}
 
     async def _insert_statistics(self) -> None:
         """Insert Opower statistics."""
-        for account in await self.api.async_get_accounts():
-            id_prefix = "_".join(
+        accounts = await self.api.async_get_accounts()
+
+        # Utility account id is not necessarily unique.  For backwards
+        # compatibility, when it is unique, use that in the id and name
+        # generation, but when it is not, use the uuid.
+        utility_account_ids = [account.utility_account_id for account in accounts]
+        duplicate_utility_account_ids = [
+            i for i in set(utility_account_ids) if utility_account_ids.count(i) > 1
+        ]
+
+        id_prefixes = {
+            account.uuid: "_".join(
                 (
                     self.api.utility.subdomain(),
                     account.meter_type.name.lower(),
-                    # Some utilities like AEP have "-" in their account id.
+                    # Some utilities like AEP have "-" in their uuid/utility_account_id.
                     # Replace it with "_" to avoid "Invalid statistic_id"
-                    account.utility_account_id.replace("-", "_"),
+                    account.uuid.replace("-", "_")
+                    if account.utility_account_id in duplicate_utility_account_ids
+                    else account.utility_account_id.replace("-", "_"),
+                )
+            )
+            for account in accounts
+        }
+        name_prefixes = {
+            account.uuid: " ".join(
+                (
+                    "Opower",
+                    self.api.utility.subdomain(),
+                    account.meter_type.name.lower(),
+                    account.uuid
+                    if account.utility_account_id in duplicate_utility_account_ids
+                    else account.utility_account_id,
                 )
             )
+            for account in accounts
+        }
+
+        for account in accounts:
+            id_prefix = id_prefixes[account.uuid]
             cost_statistic_id = f"{DOMAIN}:{id_prefix}_energy_cost"
             consumption_statistic_id = f"{DOMAIN}:{id_prefix}_energy_consumption"
```

Comment:
As discussed in the open issue, it would be great if this could be set to consumption or production depending on the account. I assume your utility website can tell which one is which to properly render them? If it isn't possible, maybe drop the consumption suffix only for duplicate utility account ids? You should also update the documentation (no need for screenshots) since for accounts like yours one has to use one statistic for the grid consumption while the other statistic for the solar production in the energy dashboard. I assume if there is no indication in the statistic name which one is which, users have to just go with one statistic and switch to the other one if it's the wrong one. That should be clear in the documentation.